packages/text-area/src/vaadin-text-area-mixin.js

```diff
@@ -209,6 +223,36 @@ export const TextAreaMixin = (superClass) =>
       }
     }
 
+    /** @private */
+    __updateMaxHeight(maxRows) {
+      if (!this._inputField || !this.inputElement) {
+        return;
+      }
+
+      if (maxRows) {
+        // For maximum height, we need to constrain the height of the input
+        // container to prevent it from growing further. For this we take the
+        // line height of the native textarea times the number of rows, and add
+        // other properties affecting the height of the input container.
+        const inputStyle = getComputedStyle(this.inputElement);
+        const inputFieldStyle = getComputedStyle(this._inputField);
+
+        const lineHeight = parseInt(inputStyle.lineHeight);
+        const contentHeight = lineHeight * maxRows;
+        const marginsAndPaddings =
+          parseInt(inputStyle.paddingTop) +
+          parseInt(inputStyle.paddingBottom) +
+          parseInt(inputStyle.marginTop) +
+          parseInt(inputStyle.marginBottom) +
+          parseInt(inputFieldStyle.paddingTop) +
+          parseInt(inputFieldStyle.paddingBottom);
```

Comment:
Could use `parseFloat` to eliminate the scrollbar caused by rounding:

https://github.com/user-attachments/assets/1631e461-b381-4711-a461-879cddfbe513

Reply:
Thanks, done.